pkgs/applications/audio/vcv-rack/default.nix

```
@@ -87,6 +101,20 @@ with lib; stdenv.mkDerivation rec {
 
     # Override the default global resource file directory
     wrapProgram $out/bin/Rack --add-flags "-s $out/share/vcv-rack"
+
+    # Extract pngs from the Apple icon image and create
+    # the missing ones from the 1024x1024 image.
+    icns2png --extract icon.icns
+    for size in 16 24 32 48 64 128 256 512 1024; do
+      mkdir -pv $out/share/icons/hicolor/"$size"x"$size"/apps
+      if [ ! -e icon_"$size"x"$size"x32.png ]
+      then
+        convert -resize "$size"x"$size" icon_1024x1024x32.png icon_"$size"x"$size"x32.png
+      fi
+      install -Dm644 icon_"$size"x"$size"x32.png $out/share/icons/hicolor/"$size"x"$size"/apps/Rack.png
+    done;
+
+    copyDesktopItems
```

```suggestion
    runHook postInstall
```

Also please add the preHook.